server/src/main/java/io/crate/expression/tablefunctions/MatchesFunction.java

```diff
@@ -19,29 +19,37 @@
  * software solely pursuant to the terms of the relevant commercial agreement.
  */
 
-package io.crate.expression.scalar.regex;
+package io.crate.expression.tablefunctions;
 
+import io.crate.common.annotations.VisibleForTesting;
 import io.crate.data.Input;
-import io.crate.expression.scalar.ScalarFunctionModule;
-import io.crate.expression.symbol.Function;
+import io.crate.data.Row;
+import io.crate.data.RowN;
+import io.crate.expression.scalar.regex.RegexMatcher;
 import io.crate.expression.symbol.Literal;
 import io.crate.expression.symbol.Symbol;
 import io.crate.expression.symbol.SymbolType;
 import io.crate.metadata.Scalar;
 import io.crate.metadata.TransactionContext;
 import io.crate.metadata.functions.Signature;
+import io.crate.metadata.tablefunctions.TableFunctionImplementation;
 import io.crate.types.DataTypes;
+import io.crate.types.RowType;
 
 import javax.annotation.Nullable;
+import java.util.Iterator;
 import java.util.List;
+import java.util.NoSuchElementException;
 
-public class MatchesFunction extends Scalar<List<String>, Object> {
+public class MatchesFunction extends TableFunctionImplementation<List<Object>> {
 
     public static final String NAME = "regexp_matches";
+    private static final RowType ROW_TYPE = new RowType(
+        List.of(DataTypes.STRING_ARRAY), List.of("groups"));
 
-    public static void register(ScalarFunctionModule module) {
+    public static void register(TableFunctionModule module) {
         module.register(
-            Signature.scalar(
+            Signature.table(
```

Comment:
Does the type signature definition still fit? Shouldn't that be a row-type now?

Reply:
It is a row_type, `new RowType(List.of(DataTypes.STRING_ARRAY), List.of("groups"))`, I followed a similar approach to GenerateSeries.